### ISSUE TITLE
type: ignore required when storing attribute to a function

### ISSUE THREAD

**Issue by timabbott** [JukkaL/mypy#1170]
Here's example code:
```
def foo():
    pass

foo.csrf_exempt = True
```
Ideally there'd be something other than a type: ignored to do here.

**Comment by gvanrossum** [comment]
Yes, function attribute are kind of wild west. I'm not sure where to
put annotations for them: often the function itself doesn't care or
know about its attributes, it's some decorator other dynamic thing
that uses a specific function attribute.

**Comment by JukkaL** [comment]
If we'd have structural subtyping, we could potentially allow defining a type that is callable + has some attributes. Then we could do something like this:

```
class MyFunc(Protocol):
    def __call__(self) -> int: ...
    attr = ... # type: bool

def _foo() -> int: ...
foo = cast(MyFunc, _foo)
foo.attr = True  # ok
foo()  # ok
```

It looks pretty ugly, though, and we still have to use a cast and refactor code.
